widget/embedded/src/store/wallets.ts

```diff
@@ -54,156 +56,189 @@ interface WalletsStore {
-        set((state) => ({
-          loading: true,
-          connectedWallets: state.connectedWallets
-            .filter((wallet) => wallet.walletType !== accounts[0].walletType)
-            .concat(
-              accounts.map((account) => {
-                const shouldMarkWalletAsSelected = !state.connectedWallets.some(
-                  (connectedWallet) =>
-                    connectedWallet.chain === account.chain &&
-                    connectedWallet.selected &&
-                    /**
-                     * Sometimes, the connect function can be called multiple times for a particular wallet type when using the auto-connect feature.
-                     * This check is there to make sure the chosen wallet doesn't end up unselected.
-                     */
-                    connectedWallet.walletType !== account.walletType
+    subscribeWithSelector((set, get) => {
+      return {
+        connectedWallets: [],
+        balances: {},
+        loading: false,
+        connectWallet: (accounts, findToken) => {
+          eventEmitter.emit(MainEvents.WalletEvent, {
+            type: WalletEventTypes.CONNECT,
+            payload: { walletType: accounts[0].walletType, accounts },
+          });
+
+          const getWalletsDetails = get().getWalletsDetails;
+          set((state) => ({
+            loading: true,
+            connectedWallets: state.connectedWallets
+              .filter((wallet) => wallet.walletType !== accounts[0].walletType)
+              .concat(
+                accounts.map((account) => {
+                  const shouldMarkWalletAsSelected =
+                    !state.connectedWallets.some(
+                      (connectedWallet) =>
+                        connectedWallet.chain === account.chain &&
+                        connectedWallet.selected &&
+                        /**
+                         * Sometimes, the connect function can be called multiple times for a particular wallet type when using the auto-connect feature.
+                         * This check is there to make sure the chosen wallet doesn't end up unselected.
+                         */
+                        connectedWallet.walletType !== account.walletType
+                    );
+                  return {
+                    balances: [],
+                    address: account.address,
+                    chain: account.chain,
+                    explorerUrl: null,
+                    walletType: account.walletType,
+                    selected: shouldMarkWalletAsSelected,
+                    loading: true,
+                    error: false,
+                  };
+                })
+              ),
+          }));
+          getWalletsDetails(accounts, findToken);
+        },
+        disconnectWallet: (walletType) => {
+          eventEmitter.emit(MainEvents.WalletEvent, {
+            type: WalletEventTypes.DISCONNECT,
+            payload: { walletType },
+          });
```

Comment:
Disconnect event for some wallets is fired multiple times. Can you check the reason?
![Screenshot from 2024-06-22 17-04-27](https://github.com/rango-exchange/rango-client/assets/99180934/4c6c330d-e872-4466-80a2-64347d58bc46)

Reply:
Fixed